xarray/core/indexing.py

```diff
@@ -1271,7 +1271,9 @@ def _decompose_outer_indexer(
                 np_indexer.append(np_slice)
             elif isinstance(k, integer_types):
                 backend_indexer.append(k)
-            elif isinstance(k, np.ndarray) and (np.diff(k) >= 0).all():
```

Comment:
k is an `np.ndarray` here unless there are duck arrays that are subclasses of it?. But ok, doesn't hurt anyway.


Or do we want to relax the isinstance check to all duck array types?

Reply:
Thanks for taking a look @headtr1ck - I agree that this is unnecessary, so I'll remove the change.